### PR TITLE
Validator client should not throw runtime error when remote beacon node not available

### PR DESCRIPTION
**Pull request by rolfyone**
BAD_GATEWAY can happen, and we should be treating it as a 'service unavailable'. Also, we should probably be logging 'service unavailable' if we ever see it, so I've added in a log warning if those occur.

Removed the response body from normal logging.

Signed-off-by: Paul Harris <paul.harris@consensys.net>

## Documentation

- [X] I thought about documentation and added the `documentation` label to this PR if updates are required.

## Changelog

- [X] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: validator/remote/src/main/java/tech/pegasys/teku/validator/remote/apiclient/ResponseHandler.java [hunk 1]
```diff
@@ -41,6 +45,7 @@ public ResponseHandler(final JsonProvider jsonProvider, final Class<T> responseC
     withHandler(SC_ACCEPTED, this::noValueHandler);
     withHandler(SC_NO_CONTENT, this::noValueHandler);
     withHandler(SC_SERVICE_UNAVAILABLE, this::noValueHandler);
+    withHandler(SC_BAD_GATEWAY, this::noValueHandler);
```
Comment (ajsutton) [inline validator/remote/src/main/java/tech/pegasys/teku/validator/remote/apiclient/ResponseHandler.java]: This feels slightly wrong to me. If we're getting errors from the beacon chain that's not really the same as `SC_NO_CONTENT` - it should be an error that we log.  I'm not really sure why we've been mapping `SC_SERVICE_UNAVAILABLE` to just an empty response either...

File: validator/remote/src/main/java/tech/pegasys/teku/validator/remote/apiclient/ResponseHandler.java [hunk 2]
```diff
@@ -64,15 +69,22 @@ public ResponseHandler(final JsonProvider jsonProvider, final Class<T> responseC
...
         String.format(
-            "Unexpected response from Beacon Node API (url = %s, status = %s, response = %s)",
-            request.url(), response.code(), response.body().string()));
+            "Unexpected response from Beacon Node API (url = %s, status = %s)",
+            request.url(), response.code()));
   }
 
   private Optional<T> noValueHandler(final Request request, final Response response) {
+    if (response.code() != SC_ACCEPTED && response.code() != SC_NO_CONTENT) {
+      LOG.warn("No value from url={}, status={}", request.url(), response.code());
```
Comment (ajsutton) [inline validator/remote/src/main/java/tech/pegasys/teku/validator/remote/apiclient/ResponseHandler.java]: I think this warning helps but "No value" isn't really what's going on.  It's really a server error.

File: validator/remote/src/main/java/tech/pegasys/teku/validator/remote/apiclient/ResponseHandler.java (hunk 1 above)
Comment (rolfyone) [reply validator/remote/src/main/java/tech/pegasys/teku/validator/remote/apiclient/ResponseHandler.java]: i agree that it should be logged, service unavailable and bad gateway should definitely be a logged output of some kind, and i settled on warn, but could be swayed to error...

**Review by ajsutton — APPROVED** [review]
LGTM.

**Pull request merged**